papermerge/core/importers/imap.py

```diff
@@ -300,12 +300,17 @@ def email_iterator(
             body,
             policy=email.policy.default
         )
+        # mark e-mail as read/seen, so that it won't be processed or imported multiple times
+        imap_client.add_flags(uid, br'\Seen')
+        
         yield email_message
 
     if delete:
+        # mark e-mail as deleted, which should lead to a deletion for most mail providers
+        # Internally, delete_messages() will just call add_flags() with the flag br'\Deleted'
+        # Note: German mail provider 'Strato' won't delete messages after specifying the deleted flag, fix incoming
         imap_client.delete_messages(messages)
```

Comment:
``delete_messages()`` will delete an email **permanently** for most mail providers. The mail won't be moved to the ``Trash`` folder. So processed mails are permanently lost if a user specifies the consumption option `IMPORT_MAIL_DELETE=True`.

**Recommendation**: Do not delete e-mails, but specify the ``SEEN`` flag and move the mail to a user's specified folder. Default folder would be `TRASH`.